### DISCUSSION TITLE
`Observe` on `IntersectionObserver` won't work on build app

### DISCUSSION THREAD

**Discussion by azzzub** [solidjs/solid#1126] [Q&A] [answered]
Hi, I try to create an infinite scrolling with solid-js using the `IntersectionObserver`, but I've found that my code is running well on the development env but it crashed on production. Here is the console log:

<img width="741" alt="image" src="https://user-images.githubusercontent.com/58354447/179806230-8244f2df-c563-4623-8aa7-6933463b8747.png">

It seems that the `ref` was `undefined` at first, but why does this only happen in the production?

[Here you can see the code](https://playground.solidjs.com/?version=1.4.1#NobwRAdghgtgpmAXGGUCWEwBowBcCeADgsrgM4Ae2YZA9gK4BOAxiWGjIbY7gAQi9GcCABM4jXgF9eAM0a0YvADo1aAGzQiAtACsyAegDucAEYqA3EogcuPfld68Ags1Zk6jLA97MhUXHAAojIycMy4XhCOvnD+cADKaADm0GqRjgBi3Om8AFLxABo5tBAAwmqxEPSExRAAsgwQEd7xABa0hpHScgrKqhraehZWVhgBjDJQrLwACrQA1nAwJQBKcGRcEGRw9lE+jbiIvFUwJuKWexBwFLgA-EdkuIwYSReOhEIAbmgMZPe8j2eEFe3iEZHoanIRzmi2WEAAIv4oMAALoXSQjaxNcSTaYwpYlRG4KC7RzQeAPJ4vN68JhqSlAkEQDEQKwyegQcI-KIASQgMgwaAC8V86jUAB4ACpYXgAVV41wCojIeUKADpAhV4E0AHwACg+tEIZCOIG8sWYrSOfhEJTU+F4ktRvAAPrTRHABVcRK7jhC1L7JmptjTLWg1CIhBAjnqhUsjtLeBgxBQji43B5xSczowdQBKXgAXh1cppVxuMYLxd4UAg+BprSgZAaQiOJlo6kqNLUtCgIhedXWZCgSTg-3yBQ1WuEuBpwhEg-cI7HRwnU6WM5pZFFajUktaYPaEf+2fOVkkBbNewqfFoJm2jE+4jWMiOtfrmOiJUevGAPb7LwytsuAADK9v2wIokWPh+MKySpHqQbbHmFzeCUDQcrgep6lWJZXo4X5bLe97iE+EiFsccCGLwfLjNsXIlAA8iRj7iNhM7POsuGkgRjhoDIvB6hxaDrMAAAMKJqmgZC0eI9G4C8l7eLxfECQa8jGmqjbNtwcBKXsKkEcBYEAcCepPPQek0oZ7waWQarllhKHKSpLKGdIcDBjs+E2cZ4EvIhUBec5BkEW5YUhS5d4PmRarRaRcB6vFrGMC+kWhSU5SVNU2HcT5vHJbFHKFYlJWpZ6IW8ReNLVZ++xEbwZXMToYS4AAEnAyStHwFEAISGpp27yLu+6HuoIgubcfQAKxiYQFAqC5RwDfZQ1iqN6xHj6ADUfTzcMrJ7DEcTBKE4S5UWeEucw36dmqPZJElLFkWlNXpdEsFBCErUXdW+VJmp-V2VpTYtnpPG8X5pmPUhVkueFtWHY4Qi4EwUR6i54o6i5jjilkEgWq0hYgCtaqE9IYYRlGxOk5TkbCNI+jY6FuP9p8OMEUIMjE2VL7hYZjz4BUxP-YZrSdUk3VHE1JgteEHVdc0LMqSoqCMEkGBaLgRoqEcKhaCovC7TLcvtRL3U5O5-MEUzHPim0HQc44hjixAxP-hBj15tbvF09TJPAx7A5DsurpuuKbM6iZntqrH4r6JHPu8Lbyv2+01Eu8IxNA0a9naWDFOtOG9NuwHudk6Ii7DqOjPM7x8d17wIUsmyHIMVEpQHGx+mET+wCEMuQFwLgMzLlBFHHXBKRBXqYmVfVfciEiQ+4ESUDjzBsRT6k4r4nCa+ovqqLzzdDXAPnukr21oO6Rvk8JPBM8WXDSML3woS4Jaa-X6IFTkTWZB8CckEnla634+AaBgEKaCABGMS1k36NRCMBaCA9Rw4V4AAKl4JA6Bu0YFG14DA4h8CXJPAdKLRBYJoJQEMOgd+w9LQY2Vo4AABt1XAxpED6H0FwRYUBCBoDVDdfQAi0D6E+AAJl4QsAkEBbi4NwIWAAJCARRkgABktBkHDxUSAbRMhgKSFYRzeevtwG8AAPo0LodAsEao9AlBwgg1SglLE9xsqfH8VxDBr2gtYpssxZFwjWBsb8cAXEEX4oJHxa81RgghOQe6wgki4FaLwcUOCOBCg8TZAEw9r46SEIFYKkTHBJ0cMBNe2F3GXV-LHNUliZQNNiUieJ6xElkBRGYsK8MfD+EtIJcQ8hGBvjrLkgiXi7rDO4EJRgIy1TwCXKOHp3QMBBXtBDIyw9R7oLQYlAs+DVneEkKhPYKM0aCUxo3eufIvRxhFMNNQTsFRTCJiAJexIcIVMojcYmGC-ovMcDw3g8QezUSuLgQw3B5i8EMEKdJwFJQcDgAwXAQL8m4GRfANFv0rosJUh-L+SIf4iD-s4jFkgZRwLEnPMpUgfkXyEMTJlByfnzirsuYmEc0CfB1AAOVoLwZYQheCfKgL1eOicOY3JUiAPUnwgqWVAQS5Gw9LnMLyfXSOGL66tAAMw6hAIqtQlkHKwDgJIeOBrZVaoySSA8npiYmrNXSSQRqXVwDVG6+OUBbU2Slby-1vEelVSTvHO5gphQ7jULKhu3hm6YijGIRgeKMmd0wuIZOOoZS2mYPQbUuA1SjlwJqDcTQABC+AeQiAxmAARhAVB5l6iFMAkgURAA)

**Comment by maxmilton** [comment]
It sounds like there could be some production minification issue but it's a little tough going off the information you've provided.

Do you have a repo link you could share? Or could you create a simple reproduction in <https://playground.solidjs.com/> or another such JS REPL?

**Reply by azzzub** [reply]
Sorry for that, here I update the reproduction with solid js playground, but you can see the code won't reproduce the error because it is in the development environment, you can see the error after building the app

[Repro](https://playground.solidjs.com/?version=1.4.1#NobwRAdghgtgpmAXGGUCWEwBowBcCeADgsrgM4Ae2YZA9gK4BOAxiWGjIbY7gAQi9GcCABM4jXgF9eAM0a0YvADo1aAGzQiAtACsyAegDucAEYqA3EogcuPfld68Ags1Zk6jLA97MhUXHAAojIycMy4XhCOvnD+cADKaADm0GqRjgBi3Om8AFLxABo5tBAAwmqxEPSExRAAsgwQEd7xABa0hpHScgrKqhraehZWVhgBjDJQrLwACrQA1nAwJQBKcGRcEGRw9lE+jbiIvFUwJuKWexBwFLgA-EdkuIwYSReOhEIAbmgMZPe8j2eEFe3iEZHoanIRzmi2WEAAIv4oMAALoXSQjaxNcSTaYwpYlRG4KC7RzQeAPJ4vN68JhqSlAkEQDEQKwyegQcI-KIASQgMgwaAC8V86jUAB4ACpYXgAVV41wCojIeUKADpAhV4E0AHwACg+tEIZCOIG8sWYrSOfhEJTU+F4ktRvAAPrTRHABVcRK7jhC1L7JmptjTLWg1CIhBAjnqhUsjtLeBgxBQji43B5xSczowdQBKXgAXh1cppVxuMYLxd4UAg+BprSgZAaQiOJlo6kqNLUtCgIhedXWZCgSTg-3yBQ1WuEuBpwhEg-cI7HRwnU6WM5pZFFajUktaYPaEf+2fOVkkBbNewqfFoJm2jE+4jWMiOtfrmOiJUevGAPb7LwytsuAADK9v2wIokWPh+MKySpHqQbbHmFzeCUDQcrgep6lWJZXo4X5bLe97iE+EiFsccCGLwfLjNsXIlAA8iRj7iNhM7POsuGkgRjhoDIvB6hxaDrMAAAMKJqmgZC0eI9G4C8l7eLxfECQa8jGmqjbNtwcBKXsKkEcBYEAcCepPPQek0oZ7waWQarllhKHKSpLKGdIcDBjs+E2cZ4EvIhUBec5BkEW5YUhS5d4PmRarRaRcB6vFrGMC+kWhSU5SVNU2HcT5vHJbFHKFYlJWpZ6IW8ReNLVZ++xEbwZXMToYS4AAEnAyStHwFEAISGpp27yLu+6HuoIgubcfQAKxiYQFAqC5RwDfZQ1iqN6xHj6ADUfTzcMrJ7DEcTBKE4S5UWeEucw36dmqPZJElLFkWlNXpdEsFBCErUXdW+VJmp-V2VpTYtnpPG8X5pmPUhVkueFtWHY4Qi4EwUR6i54o6i5jjilkEgWq0hYgCtaqE9IYYRlGxOk5TkbCNI+jY6FuP9p8OMEUIMjE2VL7hYZjz4BUxP-YZrSdUk3VHE1JgteEHVdc0LMqSoqCMEkGBaLgRoqEcKhaCovC7TLcvtRL3U5O5-MEUzHPim0HQc44hjixAxP-hBj15tbvF09TJPAx7A5DsurpuuKbM6iZntqrH4r6JHPu8Lbyv2+01Eu8IxNA0a9naWDFOtOG9NuwHudk6Ii7DqOjPM7x8d17wIUsmyHIMVEpQHGx+mET+wCEMuQFwLgMzLlBFHHXBKRBXqYmVfVfciEiQ+4ESUDjzBsRT6k4r4nCa+ovqqLzzdDXAPnukr21oO6Rvk8JPBM8WXDSML3woS4Jaa-X6IFTkTWZB8CckEnla634+AaBgEKaCABGMS1k36NRCMBaCA9Rw4V4AAKl4JA6Bu0YFG14DA4h8CXJPAdKLRBYJoJQEMOgd+w9LQY2Vo4AABt1XAxpED6H0FwRYUBCBoDVDdfQAi0D6E+AAJl4QsAkEBbi4NwIWAAJCARRkgABktBkHDxUSAbRMhgKSFYRzeevtwG8AAPo0LodAsEao9AlBwgg1SglLE9xsqfH8VxDBr2gtYpssxZFwjWBsb8cAXEEX4oJHxa81RgghOQe6wgki4FaLwcUOCOBCg8TZAEw9r46SEIFYKkTHBJ0cMBNe2F3GXV-LHNUliZQNNiUieJ6xElkBRGYsK8MfD+EtIJcQ8hGBvjrLkgiXi7rDO4EJRgIy1TwCXKOHp3QMBBXtBDIyw9R7oLQYlAs+DVneEkKhPYKM0aCUxo3eufIvRxhFMNNQTsFRTCJiAJexIcIVMojcYmGC-ovMcDw3g8QezUSuLgQw3B5i8EMEKdJwFJQcDgAwXAQL8m4GRfANFv0rosJUh-L+SIf4iD-s4jFkgZRwLEnPMpUgfkXyEMTJlByfnzirsuYmEc0CfB1AAOVoLwZYQheCfKgL1eOicOY3JUiAPUnwgqWVAQS5Gw9LnMLyfXSOGL66tAAMw6hAIqtQlkHKwDgJIeOBrZVaoySSA8npiYmrNXSSQRqXVwDVG6+OUBbU2Slby-1vEelVSTvHO5gphQ7jULKhu3hm6YijGIRgeKMmd0wuIZOOoZS2mYPQbUuA1SjlwJqDcTQABC+AeQiAxmAARhAVB5l6iFMAkgURAA)

**Comment by azzzub** [comment, marked as answer]
I catch the issue, it's because I create the infinite scroll component outside of the solidjs app folder and try to call that component inside of my solidjs app. Even using `npm link`, I still get the issue on production only.
But for now, I will move that component inside of my app folder and it's looking good on production.
Have a plan to publish the component on npm package.
